src/modm/driver/inertial/lsm6dso.hpp

```diff
@@ -0,0 +1,288 @@
+		CTRL1_OIS = 0x70,
+		CTRL2_OIS = 0x71,
+		CTRL3_OIS = 0x72,
+		X_OFS_USR = 0x73,
+		Y_OFS_USR = 0x74,
+		Z_OFS_USR = 0x75,
+		FIFO_DATA_OUT_TAG = 0x78,
+		FIFO_DATA_OUT_X_L = 0x79,
+		FIFO_DATA_OUT_X_H = 0x7A,
+		FIFO_DATA_OUT_Y_L = 0x7B,
+		FIFO_DATA_OUT_Y_H = 0x7C,
+		FIFO_DATA_OUT_Z_L = 0x7D,
+		FIFO_DATA_OUT_Z_H = 0x7E,
+	};
+
+	enum class
+	LinearRange : uint8_t
+	{
+		// assumes that XL_FS_MODE==0 in register CTRL8_XL
+		G2  = (0b00 << 2),
+		G4  = (0b10 << 2),
+		G8  = (0b11 << 2),
+		G16 = (0b01 << 2),
+	};
+
+	enum class
+	AngularRange : uint8_t
+	{
+		dps125  = (0b001 << 1),
+		dps250  = (0b000 << 1),
+		dps500  = (0b010 << 1),
+		dps1000 = (0b100 << 1),
+		dps2000 = (0b110 << 1),
+	};
+
+protected:
+	/// @cond
+	static constexpr uint8_t Ctrl1XlOutputDataRateMask = (0b1111 << 4);
+	static constexpr uint8_t Ctrl1XlLinearRangeMask = (0b11 << 2);
+	static constexpr uint8_t Ctrl2GOutputDataRateMask = (0b1111 << 4);
+	static constexpr uint8_t Ctrl2GAngularRangeMask = (0b111 << 1);
+
+	template<frequency_t frequency>
+	static constexpr std::pair<float, uint8_t>
+	getClosestOdr()
+	{
+		/*
+		```python3
+		available_freqs = [12.5, 26, 52, 104, 208, 416, 833, 1660, 3330, 6660]
+		for i in range(len(available_freqs) - 1):
+			l = available_freqs[i]
+			u = available_freqs[i+1]
+			print(f'else if constexpr (frequency < {round(2*l*u/(l+u))}) return ...')
+		```
+		*/
+		if constexpr (frequency == 0) { return std::make_pair(0.0/*Hz*/, 0b0000); }
+		else if constexpr (frequency < 17) { return std::make_pair(12.5/*Hz*/, 0b0001); }
+		else if constexpr (frequency < 35) { return std::make_pair(26/*Hz*/, 0b0010); }
+		else if constexpr (frequency < 69) { return std::make_pair(52/*Hz*/, 0b0011); }
+		else if constexpr (frequency < 139) { return std::make_pair(104/*Hz*/, 0b0100); }
+		else if constexpr (frequency < 277) { return std::make_pair(208/*Hz*/, 0b0101); }
+		else if constexpr (frequency < 555) { return std::make_pair(416/*Hz*/, 0b0110); }
+		else if constexpr (frequency < 1109) { return std::make_pair(833/*Hz*/, 0b0111); }
+		else if constexpr (frequency < 2216) { return std::make_pair(1660/*Hz*/, 0b1000); }
+		else if constexpr (frequency < 4440) { return std::make_pair(3330/*Hz*/, 0b1001); }
+		else { return std::make_pair(6660/*Hz*/, 0b1010); }
+	}
+
+	static constexpr uint8_t
+	i(Register v) { return uint8_t(v); }
+	static constexpr uint8_t
+	i(LinearRange v) { return uint8_t(v); }
+	static constexpr uint8_t
+	i(AngularRange v) { return uint8_t(v); }
+	/// @endcond
+}; // struct lsm6dso
+
+/**
+ * @tparam	Transport	Either the I2C or SPI Transport Layer.
+ * @see Lsm6TransportI2c
+ * @see Lsm6TransportSpi
+ *
+ * @ingroup modm_driver_lsm6dso
+ * @author	Raphael Lehmann
+ */
+template < class Transport >
+class Lsm6dso : public lsm6dso, public Transport
+{
+public:
+	/**
+	 * @brief Constructor
+	 *
+	 * For I2c this also sets the address to 0x6a (alternative: 0x6b with pin
+	 * SA0 high).
+	 */
+	Lsm6dso(uint8_t address=0x6a);
```

Comment:
```suggestion
	explicit Lsm6dso(uint8_t address=0x6a);
```

[C.46: By default, declare single-argument constructors explicit](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-explicit)